webapp/lib/charts.dart

```diff
@@ -210,16 +234,55 @@ class SystemEventsTimeseriesLineChartView extends DailyTimeseriesLineChartView {
         ],
         yAxes: [
           new chartjs.ChartYAxe()
-            ..ticks = (new chartjs.LinearTickOptions()
-              ..beginAtZero = true
-              ..max = 2)
-              ..display = false
-              ..gridLines = (new chartjs.GridLineOptions(zeroLineWidth: 0))
+            ..ticks = (new chartjs.LinearTickOptions()..beginAtZero = true)
+            ..display = false
+            ..gridLines = (new chartjs.GridLineOptions(zeroLineWidth: 0))
         ]),
       hover: new chartjs.ChartHoverOptions()..animationDuration = 0
     );
 
     var chartConfig = new chartjs.ChartConfiguration(type: 'line', data: chartData, options: chartOptions);
     chart = chartjs.Chart(canvas.getContext('2d'), chartConfig);
   }
+
+  void updateChart(Map<String, Map<DateTime, num>> updatedCountsAtTimestampList, {String timeScaleUnit = 'day', num upperLimit}) {
+    // Clearing up previous data
+    chartData.datasets.clear();
+
+    // Show new data
+    updatedCountsAtTimestampList.forEach((datasetLabel, data) {
+      List<chartjs.ChartPoint> timeseriesPoints = [];
+      List<DateTime> sortedDateTimes = data.keys.toList()
+        ..sort((t1, t2) => t1.compareTo(t2));
+      for (var datetime in sortedDateTimes) {
+        var value = data[datetime];
+        timeseriesPoints.add(
+            new chartjs.ChartPoint(t: datetime.toIso8601String(), y: value));
+      }
+      var newChartDataset = new chartjs.ChartDataSets(
+        label: datasetLabel,
+        backgroundColor: '${_stringToHexColor(datasetLabel)}4D',
+        borderColor: _stringToHexColor(datasetLabel),
+        data: [],
+        showLine: false,
+        pointRadius: 8,
+        hoverRadius: 8);
+      newChartDataset.data.addAll(timeseriesPoints);
+      chartData.datasets.add(newChartDataset);
+    });
+    var timeScaleOptions = new chartjs.TimeScale(unit: timeScaleUnit);
+    if (timeScaleUnit == 'hour') {
+      timeScaleOptions.stepSize = 2;
+    }
+    chart.options.scales.xAxes[0].time = timeScaleOptions;
+    if (upperLimit != null) {
+      chart.options.scales.yAxes[0].ticks = (new chartjs.LinearTickOptions()
+                                              ..beginAtZero = true
+                                              ..max = upperLimit);
+    }
+    chart.update(new chartjs.ChartUpdateProps(duration: 0));
+  }
+
+  // Alpha 100%: FF 87%: DE70%: B3 54%: 8A 50%: 80 38%: 61 12%: 1F
```

Comment:
Sorry, I meant to delete this in my previous PR as it's no longer needed. Can we do it here?
```suggestion
```